moo/src/routers/articleRouter.js

```diff
@@ -0,0 +1,138 @@
+import express from 'express';
+import prisma from '../client/prismaClient.js';
+import { assert } from 'superstruct';
+import { CreateArticle, PatchArticle, CreateComment, PatchComment } from '../validators/structor.js';
+import asyncHandler from "../middlewares/asyncHandler.js";
+
+
+const router = express.Router();
+
+router.get('/', asyncHandler(async (req, res) => {
+    const { page = 0, pageSize = 5, order = "oldest", keyword = "", } = req.query;
+    let orderBy;
+    switch (order) {
+        case "recent":
+            orderBy = { createdAt: "desc" };
+            break;
+        case "oldest":
+        default:
+            orderBy = { createdAt: "asc" };
+            break;
+    }
+    const whereClause = keyword
+    ? {
+        OR: [
+            { title: { contains: keyword, mode: "insensitive" } },
+            { content: { contains: keyword, mode: "insensitive" } },
+        ],
+    }
+    : {};
+    const articles = await prisma.article.findMany({
+        where: whereClause,
+        select: {
+            id: true,
+            title: true,
+            content: true,
+            createdAt: true,
+        },
+        orderBy,
+        skip: parseInt((page - 1) * pageSize),
```

Comment:
이거 page 0이면 음수되니까 다음과 같이 고쳐주세요. 또는 page 검증 로직에서 1이상으로 만들어주셔도 되고요
```suggestion
        skip: parseInt(Math.max(0, (page - 1) * pageSize)),
```